onnxruntime/core/common/spin_pause.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/spin_pause.h"
+
+#if defined(_M_AMD64)
+#include <intrin.h>
+#endif
+
+#if defined(__x86_64__)
+#include <xmmintrin.h>
+#endif
+
+#if defined(_M_AMD64) || defined(__x86_64__)
+#include "core/common/cpuid_info.h"
+#if defined(__linux__)
+#include <x86intrin.h>
+#include <immintrin.h>
+#include <cstdlib>
+#endif
+#endif
+
+namespace onnxruntime {
+
+#if defined(_M_AMD64) || defined(__x86_64__)
+static const bool tpause = CPUIDInfo::GetCPUIDInfo().HasTPAUSE();
+static const std::uint64_t spin_delay_cycles = 1000;
```

Comment:
also some naming and declaration suggestions
```suggestion
static const bool has_tpause = CPUIDInfo::GetCPUIDInfo().HasTPAUSE();
static constexpr uint64_t tpause_spin_delay_cycles = 1000;
```

Reply:
@edgchen1 names are updates in latest commit, thanks! :)